### PR TITLE
#113 - introduced ValidUpload abstraction

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #113 
- introduced `ValidUpload` abstraction and `Dummy` implementation
- used this new abstraction in `UpdateMavenSlice`
- created test for `UpdateMavenSlice`
- added todos to continue

### CODE REVIEW

File: src/test/java/com/artipie/maven/http/UpdateMavenSliceTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        final byte[] body = "java metadata".getBytes();
+        final String location = "org/example/artifact/0.1/maven-metadata.xml";
+        MatcherAssert.assertThat(
+            "Returns CREATED status",
+            new UpdateMavenSlice(storage, new ValidUpload.Dummy(false)).response(
+                new RequestLine("PUT", String.format("/%s", location)).toString(),
+                Collections.emptyList(),
+                Flowable.fromArray(ByteBuffer.wrap(body))
+            ),
+            new RsHasStatus(RsStatus.CREATED)
```
Comment (g4s8) [inline src/test/java/com/artipie/maven/http/UpdateMavenSliceTest.java:84]: @olenagerasimova I think it's not correct to return `CREATED` status for failed upload, maybe we can use `400` bad request here?

Comment (olenagerasimova) [reply src/test/java/com/artipie/maven/http/UpdateMavenSliceTest.java]: @g4s8 corrected, but please note that previous implementation saved `maven-metadata.xml` responded `created` and only then started `maven-metadata.xml` update process. Now we return response after validation and update metadata process.

**Comment by olenagerasimova** [comment]
@g4s8 please take a look, corrected and answered

**Comment by 0crat** [comment]
Job `gh:artipie/maven-adapter#121` is not assigned, can't get performer

**Pull request merged**